client/app/scripts/superdesk/notification/notification.js

```diff
@@ -30,19 +34,67 @@
             return;
         }
 
-        var ws = new WebSocket(config.server.ws);
-
-        ws.onmessage = function(event) {
-            var msg = angular.fromJson(event.data);
-            $rootScope.$broadcast(msg.event, msg.extra);
-            if (_.contains(ReloadEvents, msg.event)) {
-                $rootScope.$broadcast('reload', msg);
-            }
+        var connect = function() {
+            ws = new WebSocket(config.server.ws);
+            bindEvents();
         };
 
-        ws.onerror = function(event) {
-            console.error(event);
+        var bindEvents = function() {
+            ws.onmessage = function(event) {
+                var msg = angular.fromJson(event.data);
+                $rootScope.$broadcast(msg.event, msg.extra);
+                if (_.contains(ReloadEvents, msg.event)) {
+                    $rootScope.$broadcast('reload', msg);
+                }
+            };
+
+            ws.onerror = function(event) {
+                console.error('in error ws.onerror');
+                console.error(event);
+            };
+
+            ws.onopen = function(event) {
+                clearInterval(connectTimer);
+                $rootScope.$broadcast('connected');
+            };
+
+            ws.onclose = function(event) {
+                $rootScope.$broadcast('disconnected');
+                clearInterval(connectTimer);
+                connectTimer = setInterval(function() {
```

Comment:
Could be a typo, or an intentional way of avoiding firing unnecessary $digest cycles, since all the event are propagated by `$broadcasting` them. But it is also true that this can be achieved with the `$interval` service as well, it's just a matter of passing the right parameter to it.

Without a specific reason for using `setInterval()`, it is probably slightly better to use `$interval` IMO, because it would represent a better (default) coding pattern example for future contributors. :)